### ISSUE TITLE
V0.3 feedback

### ISSUE THREAD

**Issue by damithc** [CS2103AUG2016-W10-C4/main#115]
I'm giving you guys one day to do any further updates to the user guide and developer guide (_**if**_ you want to do any further updates based on what I said during the tutorial)  before I review them on GitHub.

**Comment by damithc** [comment]
# Developer Guide
* a) Can limit this SD to high-level components only? i.e. At high level, reader should not know what's happening inside the logic component. Alternatively, you can promote Parser and Dispatcher to high-level components in which case the architecture diagram needs updating. b), limit arrows to method calls and returns. The current one is hard to follow. It looks like method don't return to the caller, but return to other parties. c) `Parser` --> `:Parser` etc.
![image](https://cloud.githubusercontent.com/assets/1673303/19549478/1b39cd5a-96d7-11e6-8adf-7089056d17bd.png)
* Something's off in this diagram. Is `Parser` the one doing the dispatching? 
![image](https://cloud.githubusercontent.com/assets/1673303/19551608/a1263dca-96de-11e6-9d30-1d7598255040.png)
* `ToDoResult` name is a bit misleading as it doesn't mention anything about being the result of parsing.
* If you split model into two sub components, give a diagram at that level (showing just those two components) before going into details of each sub component.  i.e. design the level at one level before you drill down to the lower level. The sample project has a two-level design. If you go for a three-level design, then you should explain it at three levels.
* This may be of interest in case you want to describe your design decision w.r.t model-storage coupling  http://forthescience.org/books/modelviewcontroller/04_advanced_mvc/model_persistence.html
* Suggested additions:
  * Explain how your design support undo/redo. There are multiple ways this can be done, so it will be useful to explain how you do it and why you chose to do it that way. 
  * Similarly, you can explain other interesting design decisions you made.
* May be it's time to get rid of unused code and bring up your coverage level?

# User Guide

* Based on current progress you are well on your way to meet module expectations of the project. Great job. However, if you are interested in making the product actually usable, there are more things that can be done, some even without much effort. e.g. giving shorter aliases for commands. Also, try different symbols and find the one that doesn't slow down the typing speed. Just because DOS uses `x/` doesn't mean you are stuck with it. Jim is a Windows user and a fast typist, but that doesn't mean he is used to typing DOS commands. Just don't assume he is used to typing Unix commands because that's a confirmed 'no'.
* Think how to optimize the UI further. It looks nice and polished, but is it as useful as it can be? Can it show at a whole day's schedule at a time? When you add/edit/delete, does it give visual cues to confirm what happened was exactly what user intended?
* The 'Command summary' section will be used heavily by our testers. Try to make it easy to use as possible. e.g. give examples for each command.

**Comment by cricketer94** [comment]
Issue addressed here will be closed when #152 lands.
